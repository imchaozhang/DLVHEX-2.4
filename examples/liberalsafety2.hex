% The program is safe because testSubstr returns only smaller strings wrt. to the wellordering of string length

s("helloworld").
s(Y) :- s(X), &testSubstr[X,1,10](Y).
