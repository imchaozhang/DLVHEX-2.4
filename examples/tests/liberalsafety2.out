{s("helloworld"),s("elloworld"),s("lloworld"),s("loworld"),s("oworld"),s("world"),s("orld"),s("rld"),s("ld"),s("d"),s("")}
